libass/ass_directwrite.c

```diff
@@ -604,7 +606,7 @@ static void add_font_face(IDWriteFontFace *face, ASS_FontProvider *provider,
 {
     ASS_FontProviderMetaData meta = {0};
 
-    IDWriteFontFace3 *face3;
+    IDWriteFontFace3 *face3 = NULL;
```

Comment:
FWIW we trust that if the call has `SUCCEEDED`, then it has filled this variable with some well-defined value per its contract. But this is fine too.

Reply:
Technically it should, but it's probably still a bit safer and more consistent with the rest of the file where it was generally assigned to NULL in advance. I can picture some edge case situations like stub code in Wine and such, which might return a fake success response but not do anything with the params (they often have a bunch of stubs for incomplete compatibility layers just to get a portion of apps running at all). So that's why I'd consider it a worthwhile good practice